ddtrace/contrib/internal/ray/patch.py

```diff
@@ -0,0 +1,439 @@
+            f"exec {entrypoint_name}", service=os.environ.get(RAY_JOB_NAME), span_type=SpanTypes.RAY
+        ) as span:
+            span.set_tag_str(SPAN_KIND, SpanKind.CONSUMER)
+            _inject_ray_span_tags_and_metrics(span)
+
+            return method(self, *args, **kwargs)
+
+    return _traced_exec_entrypoint_method
+
+
+@contextmanager
+def _trace_actor_method(self: Any, method: Callable[..., Any], dd_trace_ctx, *args, **kwargs):
+    context = tracer.context_provider.active()
+    if context is None:
+        context = _TraceContext._extract(dd_trace_ctx)
+
+    with long_running_ray_span(
+        f"{self.__class__.__name__}.{method.__name__}",
+        service=os.environ.get(RAY_JOB_NAME),
+        span_type=SpanTypes.RAY,
+        child_of=context,
+        activate=True,
+    ) as actor_execute_span:
+        set_maybe_big_tag(actor_execute_span, "ray.actor_method.args", args)
+        set_maybe_big_tag(actor_execute_span, "ray.actor_method.kwargs", kwargs)
+
+        yield actor_execute_span
+
+
+def _inject_tracing_actor_method(method: Callable[..., Any]) -> Any:
+    def _traced_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
+
+        if _dd_trace_ctx is None and tracer.current_span() is None:
+            return method(self, *args, **kwargs)
+
+        with _trace_actor_method(self, method, _dd_trace_ctx, *args, **kwargs):
+            return method(self, *args, **kwargs)
+
+    return _traced_method
+
+
+def _inject_tracing_async_actor_method(method: Callable[..., Any]) -> Any:
+    async def _traced_async_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
+
+        if _dd_trace_ctx is None and tracer.current_span() is None:
+            return await method(self, *args, **kwargs)
+
+        with _trace_actor_method(self, method, _dd_trace_ctx, *args, **kwargs):
+            return await method(self, *args, **kwargs)
+
+    return _traced_async_method
+
+
+def inject_tracing_into_actor_class(wrapped, instance, args, kwargs):
+    cls = wrapped(*args, **kwargs)
+    module_name = str(cls.__module__)
+    class_name = str(cls.__name__)
+
+    # Skip tracing for certain ray modules
+    if module_name.startswith("ray.dag") or module_name.startswith("ray.experimental"):
+        return cls
+
+    # Determine if the class is a JobSupervisor
+    is_job_supervisor = f"{module_name}.{class_name}" == "ray.dashboard.modules.job.job_supervisor.JobSupervisor"
+    # We do not want to instrument ping and polling to remove noise
+    methods_to_ignore = {"ping", "_polling"} if is_job_supervisor else set()
+
+    methods = inspect.getmembers(cls, is_function_or_method)
+    for name, method in methods:
+        if name in methods_to_ignore:
+            continue
+
+        if (
+            is_static_method(cls, name)
+            or is_class_method(method)
+            or inspect.isgeneratorfunction(method)
```

Comment:
I think we have a ddtrace helper for this. Not 100% sure tho. Check the implementation of tracer.wrap